### PR TITLE
remove loop dependency

### PR DESCRIPTION
**Pull request by joostjager**
PR #1 left a back-reference to `loop` in place. This is cleaned up in this pr.

### CODE REVIEW

File: network.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
+package lndclient
+
+import (
+	"errors"
+
+	"github.com/btcsuite/btcd/chaincfg"
+)
+
+// Network defines the chain that we operate on.
+type Network uint8
```
Comment (guggero) [inline network.go:10]: Why not use a string type directly?

File: tx_utils.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
+package lndclient
+
+import (
+	"bytes"
+
+	"github.com/btcsuite/btcd/wire"
+)
+
+// encodeTx encodes a tx to raw bytes.
+func encodeTx(tx *wire.MsgTx) ([]byte, error) {
```
Comment (guggero) [inline tx_utils.go:10]: Might as well export these as we're building a library anyway.

File: network.go [hunk 3]
```diff
@@ -23,6 +23,26 @@ const (
 	NetworkSimnet
 )
 
+// NetworkFromStr parses a network from string.
+func NetworkFromStr(text string) (Network, error) {
+	switch text {
+	case "mainnet":
+		return NetworkMainnet, nil
+
+	case "testnet":
```
Comment (guggero) [inline network.go:32]: Add `testnet3` as well to be compliant with some notations?

File: lnd_services.go [hunk 4]
```diff
@@ -127,25 +126,25 @@ func NewLndServices(cfg *LndServicesConfig) (*GrpcLndServices, error) {
 	macaroonDir := cfg.MacaroonDir
 	if macaroonDir == "" {
 		switch cfg.Network {
-		case "testnet":
+		case NetworkTestnet:
```
Comment (guggero) [inline lnd_services.go:129]: Couldn't this whole switch statement now be removed as `cfg.Network` is now guarded against being an unknown network?

File: network.go (hunk 1 above)
Comment (joostjager) [reply network.go]: It isn't the same abstraction, but works too. Changed.

File: tx_utils.go (hunk 2 above)
Comment (joostjager) [reply tx_utils.go:10]: I think this is functionality to belongs more in `github.com/btcsuite/btcd/wire`

File: network.go (hunk 3 above)
Comment (joostjager) [reply network.go]: It doesn't match the network returned by `lnd.GetInfo`. Don't really want to add anything we don't need now.

File: lnd_services.go (hunk 4 above)
Comment (joostjager) [reply lnd_services.go:129]: It isn't guarded any more now, because we move to string

**Pull request merged**